### ISSUE TITLE
Document how to use OAuth2 with Azure AD

### ISSUE THREAD

**Issue by pmlopes** [vert-x3/issues#144]
Vert.x can be easily deployed on Azure so it makes sense to document how to integrate with Azure Active Directory OAuth2.

This should be just a documentation task and verify if the generic implementation works out of the box with it.

**Comment by apatrida** [comment]
Some things I remember about this when coding it to work with Scribe:

Login scope is `User.Read`

Token URL is form of `https://login.microsoftonline.com/${accountId}/oauth2/token`
Token HTTP verb is `POST`

Authorization URL is form of `https://login.microsoftonline.com/${accountId}/oauth2/authorize?client_id=${OAuthEncoder.encode(apiKey.substringBefore('@'))}&scope=${scopeList}&response_type=code&redirect_uri=${registeredRedirectUrl}`

scopes are `,` delimited

If state is used throughout auth, the query parameter of Authorization URL is `state`

I remember something about `redirect_uri` parameter not working if it was not last thing in Authorization URL, and maybe because if you URL encoded it then it didn't work (maybe fixed in later versions of Azure AD).  So if you have mystery errors there with that redirect URL, check that. 

When coding against Scribe library had to specify 

```
 override fun getAccessTokenExtractor(): AccessTokenExtractor? {
        return JsonTokenExtractor()
}
```

ALso with Scribe we had to:

```
  override public fun createService(config: OAuthConfig): OAuthService {
        return SharePointOAuthServiceImpl(this, config)
    }
```

which was implemented as:

```
open class SharePointOAuthServiceImpl(api: DefaultApi20, config: OAuthConfig) : OAuth20ServiceImpl(api, config) {
    protected override fun <T : AbstractRequest> createAccessTokenRequest(verifier: Verifier, request: T): T {
        super.createAccessTokenRequest(verifier, request)
        request.addParameter("resource", "https://graph.microsoft.com")
        request.addParameter(OAuthConstants.GRANT_TYPE, "authorization_code")
        return request
    }
}
```

Hope that helps grease the process.  I remember it being the oddest one out of 10 we implemented.

The resource I think is required you provide one, or maybe it was because we call graph API to get user info from `https://graph.microsoft.com/v1.0/me`
